core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -182,18 +180,25 @@ private[op] object FeatureDistribution {
    * @param values values to bin
    * @param sum summary info for feature (max and min)
    * @param bins number of bins to produce
-   * @param hasher hasing function to use for text
    * @return the bin information and the binned counts
    */
   // TODO avoid wrapping and unwrapping??
   private def histValues(
     values: ProcessedSeq,
     sum: Summary,
-    bins: Int,
-    hasher: HashingTF
+    bins: Int
   ): (Array[Double], Array[Double]) = {
     values match {
-      case Left(seq) => Array(sum.min, sum.max) -> hasher.transform(seq).toArray // TODO use summary info to pick hashes
+      case Left(seq) => {
+        val minBins = bins
+        val maxBins = MaxBins
+        val numBins = math.min(math.max(bins, sum.max/AvgBinValue), maxBins).toInt
```

Comment:
`sum.max / AvgBinValue`

Reply:
`.toInt` perhaps specify the explicit rounding instead.

Reply:
fixed! thanks!!